[Docs] Add the K8s agent set up document (#6184)
flyteorg/flyte: docs/deployment/agents/k8sservice.rst

.. _deployment-agent-setup-k8sservice:

Kubernetes (K8s) Service Agent
==================

> The Kubernetes (K8s) Service Agent enables machine learning (ML) users to efficiently handle non-training tasks—such as data loading, caching, and processing—concurrently with training jobs in Kubernetes clusters.

Reviewer: ```suggestion
The Kubernetes (K8s) Data Service Agent enables machine learning (ML) users to efficiently handle non-training tasks—such as data loading, caching, and processing—concurrently with training jobs in Kubernetes clusters.
```
The reason  I suggest this is to mitigate ambiguity. K8s itself is a plugin for flytepropeller, so having a "K8s Service Agent" could be misinterpreted as an Agent-equivalent

This capability is particularly valuable in deep learning applications, such as those in Graph Neural Networks (GNNs).

> This guide offers a comprehensive overview of setting up the K8s Service Agent within your Flyte deployment.

Reviewer: ```suggestion
This guide offers a comprehensive overview of setting up the K8s Data Service Agent within your Flyte deployment.
```

    If you've installed Flyte using the
    `flyte-core helm chart <https://github.com/flyteorg/flyte/tree/master/charts/flyte-core>`__, please ensure:

    * You have the correct kubeconfig and have selected the correct Kubernetes context.
    * You have configured the correct flytectl settings in ``~/.flyte/config.yaml``.

.. note::

  Add the Flyte chart repo to Helm if you're installing via the Helm charts.

  .. code-block:: bash

    helm repo add flyteorg https://flyteorg.github.io/flyte

Specify agent configuration
----------------------------

Enable the K8s service agent by adding the following config to the relevant YAML file(s):

.. code-block:: yaml

  tasks:
    task-plugins:
      enabled-plugins:
        - agent-service
      default-for-task-types:
        - dataservicetask: agent-service

.. code-block:: yaml

  plugins:
    agent-service:
      agents:
        k8sservice-agent:
          endpoint: <AGENT_ENDPOINT>
          insecure: true
      agentForTaskTypes:
      - dataservicetask: k8sservice-agent
      - sensor: k8sservice-agent

Substitute ``<AGENT_ENDPOINT>`` with the endpoint of your MMCloud agent.

Setup the RBAC
----------------------

> The k8s agent will create statefulset and expose the service end point for the statefulset pods.

Reviewer: ```suggestion
The K8s Data Service Agent will create a StatefulSet and expose the Service endpoint for the StatefulSet pods.
```

> RBAC needs to be set up to allow the K8s agent to CRUD statefulset and service.

Reviewer: ```suggestion
RBAC needs to be set up to allow the K8s Data Service Agent to perform CRUD operations on the StatefulSet and Service.
```
